mlir/lib/Tools/mlir-translate/Translation.cpp

```diff
@@ -144,6 +144,10 @@ TranslateFromMLIRRegistration::TranslateFromMLIRRegistration(
         DialectRegistry registry;
         dialectRegistration(registry);
         context->appendDialectRegistry(registry);
+        // Ensure all registered dialects are loaded
+        for (const auto &dialectName : registry.getDialectNames()) {
+          context->getOrLoadDialect(dialectName);
+        }
```

Comment:
That does not seem like the right fix indeed, the parser should load on demand. 
I need to reproduce to figure out how we get to crash.

Reply:
This PR contains a reproducer. Remove those lines and the test dialect to LLVM translation will segfault.